### PR TITLE
Third party lib clarification

### PR DESCRIPTION
**Pull request by dotproto**
Add the relevant policy text to the third party library usage documentation.

This PR also introduces styling for `<blockquote>` elements. Before this PR there was only one preexisting usage of `<blockquote>` or `>` in the documentation: the [destroyed actor](https://extensionworkshop.com/documentation/develop/known-issues/#android-destroyed-actor) section of the known issues page. That page should be visually unaffected by this change.

### CODE REVIEW

File: src/content/documentation/publish/third-party-library-usage.md
Comment (rebloor) [inline src/content/documentation/publish/third-party-library-usage.md:18]:
```suggestion
> Only release versions of third-party libraries and frameworks can be included in an add-on. Modifications to included libraries and frameworks are not permitted.
```

File: src/content/documentation/publish/third-party-library-usage.md
Comment (rebloor) [inline src/content/documentation/publish/third-party-library-usage.md:20]:
```suggestion
> See [Development Practices](/documentation/publish/add-on-policies/#development-practices) and [Add-ons Policies](/documentation/publish/add-on-policies/) for more information.
```

File: src/content/documentation/publish/third-party-library-usage.md
Comment (rebloor) [inline src/content/documentation/publish/third-party-library-usage.md:22]:
```suggestion
To complete the [addons.mozilla.org](https://addons.mozilla.org) (AMO) review process, reviewers must be able to verify the code in your extension. If you include third-party libraries in your extension, you must provide links to the library source code as part of the AMO submission process.
```

File: src/content/documentation/publish/add-on-policies.md
Comment (rebloor) [inline src/content/documentation/publish/add-on-policies.md:137]:
```suggestion
- Only release versions of third-party libraries and frameworks can be included in an add-on. Modifications to included libraries and frameworks are not permitted. See the [third-party library guidelines](/documentation/publish/third-party-library-usage/) for more information. {% comment %}Changes to this line should also be copied to /src/content/documentation/publish/third-party-library-usage.md{% endcomment %}
```

**Review by rebloor — COMMENTED** [review]
@dotproto some edits for your consideration

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [reply src/content/documentation/publish/third-party-library-usage.md:18]: Instead of duplicating the sentence, can we link to the policies section like:

`Please note the requirements for third-party libraries in the #Add-on Policies#`, where `#Add-on Policies`links to the development section of the policies.

File: src/content/documentation/publish/add-on-policies.md
Comment (wagnerand) [reply src/content/documentation/publish/add-on-policies.md:137]: Let's avoid changes to the policies page. Also, if we avoid duplicating the sentence as I suggested in my other comment, the comment is not needed.

Comment (dotproto) [reply src/content/documentation/publish/add-on-policies.md:137]: @rebloor, I'm closing this suggestion because I'm going to revert the change to the policy page per @wagnerand's [request](https://github.com/mozilla/extension-workshop/pull/1918/files#r1686573033).

File: src/content/documentation/publish/third-party-library-usage.md
Comment (dotproto) [reply src/content/documentation/publish/third-party-library-usage.md:18]: Sounds good. I'll update the PR to integrate this feedback after I merge @rebloor's other feedback.

**Comment by dotproto** [comment]
I've removed the exact policy text from the Third Party Library Usage docs [as requested](https://github.com/mozilla/extension-workshop/pull/1918/files#r1686571131) by @wagnerand. The main motivation for this PR was to quote that passage, so without it I was compelled to rewrite some sections in order to clarify and add more details to our requirements. 

@wagnerand, please take a look to see if you're comfortable with these updates. @rebloor, please hold off on another pass until then.

File: src/content/documentation/develop/known-issues.md
Comment (wagnerand) [inline src/content/documentation/develop/known-issues.md:77]: Unrelated drive-by? Mkay  :)

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [inline src/content/documentation/publish/third-party-library-usage.md:21]: This is still duplicating content from the policies page. We should avoid that because we can't re-check all related pages when the policies are updated to find copies of provisions that were updated. Wouldn't a link to the policies section be enough, as you do in line 18?

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [inline src/content/documentation/publish/third-party-library-usage.md:42]: While there are many options, in a best practices document let's aim to pick the best practice. The challenge here is that there is no one-size-fits all. For example, if there is no other reason to attach source code, we don't want the developer to do it just for third-party libraries.

In general, notes to reviewers is fine, unless the amount of libraries (and other reviewer information) is too high to be reasonable for a "Notes" field.

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [inline src/content/documentation/publish/third-party-library-usage.md:106]:
```suggestion
Reviewers must be be able to retrieve and review all packages used by your extension, so the use of private packages or non-public registries is not recommended. If you do, include those as "pre-installed" dependencies in your `node-modules` folder. Access information should be provided along with build instructions as part of the [source code submission](https://extensionworkshop.com/documentation/publish/source-code-submission/) process.
```

I am not sure if "pre-installed" is the right term here or descriptive enough. Thoughts on that wording?

**Review by wagnerand — CHANGES_REQUESTED** [review]

File: src/content/documentation/publish/third-party-library-usage.md
Comment (dotproto) [reply src/content/documentation/publish/third-party-library-usage.md:21]: No, I don't think so. This PR was created in response to a request from @mixedpuppy to quote the policy text on this page. The initial motivation for that request was that this page did not communicate the policy requirements to the reader despite the positioning that it is authoritative on the topic.

The other approach I was considering was to move the relevant policy text out of `addons-policy.md` and into a [partial](https://liquidjs.com/tutorials/options.html#Partials-Layouts). That partial would then be imported in both `addons-policy.md` and `third-party-library-usage.md`. But since we don't have an existing pattern of using partials like this I held off.

File: src/content/documentation/develop/known-issues.md
Comment (dotproto) [reply src/content/documentation/develop/known-issues.md:77]: Yeah, in retrospect this and the style change should have been a separate PR. I'll break them out now.

File: src/content/documentation/publish/third-party-library-usage.md
Comment (dotproto) [reply src/content/documentation/publish/third-party-library-usage.md:106]: I don't think "pre-installed" will quite get the idea across. I took another shot at making this passage more explicit. WDYT?

```suggestion
Reviewers must be able to retrieve and review all packages used by your extension. As a result, the use of private packages or non-public registries is permissible but not recommended. If you use non-public dependencies, you will either need to include the relevant `node_modules` sub-directories in your [source code submission](/documentation/publish/source-code-submission/) or include access credentials and instructions in your build instructions.
```

File: src/content/documentation/publish/third-party-library-usage.md
Comment (dotproto) [reply src/content/documentation/publish/third-party-library-usage.md:42]: Makes sense. In that case I think we can simplify this text a bit. WDYT?

```suggestion
Links to third party libraries must be provided when submitting a new version to AMO. You can add the links to the "Notes for Reviewers" section of your extension's details.
```

Comment (rebloor) [reply src/content/documentation/publish/third-party-library-usage.md:42]:
```suggestion
When submitting a version to AMO, links to third-party libraries must be provided. You can add the links to the "Notes for Reviewers" section of your extension's details. 
```

File: src/content/documentation/publish/third-party-library-usage.md
Comment (rebloor) [reply src/content/documentation/publish/third-party-library-usage.md:106]:
```suggestion
Reviewers must be able to retrieve and review all packages used by your extension. Therefore, the use of private packages or non-public registries is permissible but not recommended. If you use non-public dependencies, you must include the relevant `node_modules` sub-directories in your [source code submission](/documentation/publish/source-code-submission/) or access credentials and instructions in your build instructions.
```

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [reply src/content/documentation/publish/third-party-library-usage.md:21]: As I said in private, this is a best practices document, not a policies or requirements document. We should not duplicate official requirements into best practices documents. We can link to the policies where the requirements are stated, but the duplication is a recipe for failure, and we have to avoid that for content that touches policy provisions. There needs to be a single source of truth that we can we can reasonably keep up to date.

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [reply src/content/documentation/publish/third-party-library-usage.md:42]: Either is fine with me, thanks.

File: src/content/documentation/publish/third-party-library-usage.md
Comment (wagnerand) [reply src/content/documentation/publish/third-party-library-usage.md:106]: Either works for me, but let's remove the `or include access credentials and instructions in your build instructions.` part. We will need to consider and evaluate that first. Some developers have done it in the past by themselves and anecdotally, that lead to issues in reproducing the build.

File: src/content/documentation/develop/known-issues.md
Comment (dotproto) [reply src/content/documentation/develop/known-issues.md:77]: This has been moved into #1924.

**Pull request merged**
